test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/MSBuildAPIUtilityTests.cs

```diff
@@ -11,16 +13,63 @@
 using NuGet.Test.Utility;
 using NuGet.Versioning;
 using Xunit;
+using Xunit.Abstractions;
 using Project = Microsoft.Build.Evaluation.Project;
 
 namespace NuGet.CommandLine.Xplat.Tests
 {
     public class MSBuildAPIUtilityTests
     {
+        /// <summary>
+        /// Stores any exceptions that occurred when calling <see cref="MSBuildLocator.RegisterDefaults" />.
+        /// </summary>
+        private static readonly Exception MSBuildLocatorRegisterDefaultsException = null;
+
         static MSBuildAPIUtilityTests()
         {
-            MSBuildLocator.RegisterDefaults();
+            try
+            {
+                MSBuildLocator.RegisterDefaults();
+            }
+            catch (Exception e)
+            {
+                MSBuildLocatorRegisterDefaultsException = e;
+            }
+        }
+
+        public MSBuildAPIUtilityTests(ITestOutputHelper outputHelper)
+        {
+            if (MSBuildLocatorRegisterDefaultsException != null)
+            {
+                outputHelper.WriteLine("Exception when calling MSBuildLocator.RegisterDefaults(): {0}", MSBuildLocatorRegisterDefaultsException);
+            }
+
+#if !NETFRAMEWORK
+            // Print out the output from dotnet --info for debugging what's going on if this is failing for a hosted build
+            outputHelper.WriteLine("Output of 'dotnet --info':");
+            try
+            {
+                Process process = Process.Start(new ProcessStartInfo
+                {
+                    FileName = "dotnet",
```

Comment:
Shouldn't this try to run `<RepoRoot>\cli\dotnet`?

Reply:
This just helped me figure out why MSBuildLocator failed to find MSBuild.  I'm going to revert this for now and maybe send a PR to MSBuildLocator to add more diagnostic information.